startup/csx1/devices/areadetector.py

```diff
@@ -36,24 +36,108 @@ class StandardCam(SingleTrigger, AreaDetector):
     roi4 = Cpt(ROIPlugin, 'ROI4:')
     #proc1 = Cpt(ProcessPlugin, 'Proc1:')
     #trans1 = Cpt(TransformPlugin, 'Trans1:')
-
+    over1 = Cpt(OverlayPlugin, 'Over1:') ##for crosshairs in tiff
 
 class NoStatsCam(SingleTrigger, AreaDetector):
     pass
 
 
-class MonitorStatsCam(SingleTrigger, AreaDetector):
+class MonitorStatsCam(SingleTrigger, AreaDetector): #TODO does this subscribe/unsubsribe work or are we hacking EpicsSignals in custom plans
     stats1 = Cpt(StatsPlugin, "Stats1:")
     roi1 = Cpt(ROIPlugin, "ROI1:")
     proc1 = Cpt(ProcessPlugin, "Proc1:")
 
     def subscribe(self, *args, **kwargs):
-        return self.stast1.cenx.subscribe(*args, **kwargs)
+        return self.stast1.cenx.subscribe(*args, **kwargs) #TODO if this works, then add stats1.ceny too.
 
     def unbuscribe(self, *args, **kwargs):
         return self.stast1.cenx.unsubscribe(*args, **kwargs)
 
 
+def update_describe_typing(dic, obj):
+    """
+    Function for updating dictionary result of `describe` to include better typing.
+    Previous defaults did not use `dtype_str` and simply described an image as an array.
+
+    Parameters
+    ==========
+    dic: dict
+        Return dictionary of describe method
+    obj: OphydObject
+        Instance of plugin
+    """
+    key = obj.parent._image_name
+    cam_dtype = obj.parent.cam.data_type.get(as_string=True)
+    type_map = {'UInt8': '|u1', 'UInt16': '<u2', 'Float32':'<f4', "Float64":'<f8'}
+    if cam_dtype in type_map:
+        dic[key].setdefault('dtype_str', type_map[cam_dtype])
+
+
+class HDF5PluginWithFileStorePlain(HDF5Plugin_V22, FileStoreHDF5IterativeWrite): ##SOURCED FROM BELOW FROM FCCD WITH SWMR removed
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # In CSS help: "N < 0: Up to abs(N) new directory levels will be created"
+        self.stage_sigs.update({"create_directory": -3})
+        self.stage_sigs.move_to_end("create_directory", last=False)
```

Comment:
I like this solution, I hope it works as expected :)

On a side note, `.move_to_end(..., last=False)` is the worst API 😁. Good it's explained here: https://docs.python.org/3/library/collections.html#collections.OrderedDict.move_to_end. Maybe worth adding a comment above that line that we are moving the key to the beginning of the dict?

Reply:
Indeed, we'll test out today. I've added a note in https://github.com/NSLS-II-CSX/profile_collection/pull/66/commits/434e050d807aa52ce5e85ce067a64d029dde6b5b. I was definitely taking the convoluted logic for granted...